Library/Homebrew/brew.sh

```diff
@@ -347,8 +347,14 @@ case "$*" in
   --version|-v) source "${HOMEBREW_LIBRARY}/Homebrew/cmd/--version.sh"; homebrew-version; exit 0 ;;
 esac
 
+HOMEBREW_CORE_REPOSITORY_ORIGIN="$(git -C ${HOMEBREW_CORE_REPOSITORY} remote get-url origin)"
+
+# If the remote origin has been set to Homebrew/homebrew-core by the install script,
+# then we are in the case of a new installation of brew, using Homebrew/homebrew-core as a Linux core repository.
+# In that case, set HOMEBREW_FORCE_HOMEBREW_ON_LINUX=1 to set the right HOMEBREW_BOTTLE_DOMAIN below.
+
 # shellcheck disable=SC2154
-if [[ -n "${HOMEBREW_SIMULATE_MACOS_ON_LINUX}" ]]
+if [[ -n "${HOMEBREW_SIMULATE_MACOS_ON_LINUX}" || "${HOMEBREW_CORE_REPOSITORY_ORIGIN}" = "https://github.com/Homebrew/homebrew-core" ]]
```

Comment:
Maybe we should also check if we are on Linux? No need to set this env var on macOS. Let's move this into separate if statement if possible too.

Reply:
Ok makes sense, let me do that.